### PR TITLE
Use torch 2.1 and remove torch 1.13

### PR DESCRIPTION
**Pull request by vshampor**
### Changes
As stated in the title

### Reason for changes
Keeping up with the latest torch version

### Related tickets
N/A

### Tests
Existing torch test scope

### CODE REVIEW

**Review by MaximProshin — COMMENTED** [review]
@vshampor, please also update NNCF readme. 
@ksilligan , please also take a look.

**Comment by vshampor** [comment]
E2E PT build 474 shows no significant changes to the regular nightly builds

File: nncf/version.py [hunk 1]
```diff
@@ -11,6 +11,6 @@
 
 __version__ = "2.6.0"
 
-BKC_TORCH_VERSION = "2.0.1"
-BKC_TORCHVISION_VERSION = "0.15.1"
+BKC_TORCH_VERSION = "2.1.1"
```
Comment (AlexanderDokuchaev) [inline nncf/version.py:14]: Please add row "develop" to https://github.com/openvinotoolkit/nncf/blob/develop/docs/Installation.md#corresponding-versions 
with new versions of pt.

**Review by AlexanderDokuchaev — CHANGES_REQUESTED** [review]

**Comment by vshampor** [comment]
From the result visualization for the manual PTQ build 188 I can see no degradations for the lines marked as red as compared to the regular PTQ build 169.

File: nncf/version.py (hunk 1 above)
Comment (vshampor) [reply nncf/version.py:14]: Done

**Review by AlexanderDokuchaev — APPROVED** [review]

**Review by MaximProshin — APPROVED** [review]

**Review by ksilligan — APPROVED** [review]

**Pull request merged**